File: Stemmer/src/ph/bohol/util/stemmer/AffixGroup.java
package ph.bohol.util.stemmer;

import java.util.Iterator;
import java.util.LinkedList;
import java.util.Map;

public class AffixGroup
{
    private String name;
    private LinkedList<Affix> affixes = new LinkedList<Affix>();

    final void addAffix(final Affix pattern)
    {
        affixes.addLast(pattern);
    }

    final void compile(final Map<String, String> constants)
    {
        Iterator<Affix> iterator = affixes.iterator();
        while (iterator.hasNext())
        {
            iterator.next().compile(constants);
        }
    }

    public final void setName(final String newName)
    {
        name = newName;
    }

    public final String toString()
    {
        String result = "\n<group name='" + name + "'>\n";

        Iterator<Affix> iterator = affixes.iterator();
        while (iterator.hasNext())
        {
            result += iterator.next().toString();
        }
        result += ("</group>\n");
        return result;
    }

    public final LinkedList<Affix> getAffixes()
    {
        return affixes;
    }
}

File: src/stemmer/AffixGroup.ts
import {Affix, compileAffix as compileAffix, toXmlAffix as toXmlAffix} from "./Affix";

export interface AffixGroup {
    name: string;
    affixes: Affix[];
}

export function compileGroup(group: AffixGroup, constants: {[s: string]: string}): void {
    group.affixes.forEach(af => compileAffix(af, constants));
}

export function toXmlGroup(group: AffixGroup): string {
    const result = ["\n<group name='" + group.name + "'>\n"];
    group.affixes.forEach(affix => result.push(toXmlAffix(affix)));
    result.push("</group>\n");
    return result.join("");
}
